src/neptune/internal/utils/disk_utilization.py

```diff
@@ -0,0 +1,104 @@
+#
+# Copyright (c) 2023, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+__all__ = ["ensure_disk_not_overutilize"]
+
+
+import os
+from functools import wraps
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Optional,
+    Tuple,
+)
+
+import psutil
+from psutil import Error
+
+from neptune.common.warnings import (
+    NeptuneWarning,
+    warn_once,
+)
+from neptune.constants import NEPTUNE_DATA_DIRECTORY
+from neptune.envs import (
+    NEPTUNE_MAX_DISK_UTILIZATION,
+    NEPTUNE_NON_RAISING_ON_DISK_ISSUE,
+)
+
+
+def get_neptune_data_directory() -> str:
+    return os.getenv("NEPTUNE_DATA_DIRECTORY", NEPTUNE_DATA_DIRECTORY)
+
+
+def get_disk_utilization_percent(path: Optional[str] = None) -> Optional[float]:
+    try:
+        if path is None:
+            path = get_neptune_data_directory()
+
+        return float(psutil.disk_usage(path).percent)
+    except (ValueError, TypeError, OSError, Error):
+        return None
+
+
+def get_max_disk_utilization_from_env() -> Optional[float]:
+    env_limit_disk_utilization = os.getenv(NEPTUNE_MAX_DISK_UTILIZATION)
+
+    if env_limit_disk_utilization is None:
+        return None
+
+    try:
+        limit_disk_utilization = float(env_limit_disk_utilization)
+        assert 0 < limit_disk_utilization <= 100
```

Comment:
As far as I know, using `assert` statements as an exception handling method is a bad practice. Can we change it to raise `ValueError` if the conditions are not met?

Reply:
As I see we are using `assert` in many different places. E.g. https://github.com/neptune-ai/neptune-client/blob/3bab4d6336d53af8d054962b8e44e991956a6c53/src/neptune/metadata_containers/model_version.py#L282

Reply:
```
if limit_disk_utilization <= 0 or limit_disk_utilization > 100:
    raise ValueError
```

then it will be caught and handled in the `except clause`. What do you think?

Reply:
If someone launches our code with the `-o` flag then the `asserts` are not executed, which is dangerous
https://stackoverflow.com/questions/2830358/what-are-the-implications-of-running-python-with-the-optimize-flag

Reply:
Done